js/alert_banner.js

```diff
@@ -4,31 +4,108 @@
  */
 
 (function ($, Drupal, once) {
-
   'use strict';
 
-  // Function taken from https://github.com/onury/invert-color
-  function getLuminance(c) {
-    var i, x;
-    var a = []; // So we don't mutate.
-    for (i = 0; i < c.length; i++) {
-      x = c[i] / 255;
-      a[i] = x <= 0.03928 ? x / 12.92 : Math.pow((x + 0.055) / 1.055, 2.4);
+  /**
+   * Calculate relative luminance for color contrast.
+   * Function taken from https://github.com/onury/invert-color
+   *
+   * @param {string|Array} color - The color value (hex or RGB array).
+   * @return {number} The relative luminance value.
+   */
+  function getLuminance(color) {
+    let r, g, b;
+
+    // Handle RGB array
+    if (Array.isArray(color)) {
+      [r, g, b] = color.map(x => parseInt(x) / 255);
+    }
+    // Handle RGB string
+    else if (color.indexOf('rgb') >= 0) {
+      [r, g, b] = color.replace(/rgb\(|\)|\s/gi, '').split(',').map(x => parseInt(x) / 255);
+    }
+    // Handle hex color
+    else if (typeof color === 'string' && color.startsWith('#')) {
+      const hex = color.replace('#', '');
+      r = parseInt(hex.substr(0, 2), 16) / 255;
+      g = parseInt(hex.substr(2, 2), 16) / 255;
+      b = parseInt(hex.substr(4, 2), 16) / 255;
     }
+    // Default to black if color format is not recognized
+    else {
+      return 0;
+    }
+
+    // Calculate luminance using WCAG formula
+    const a = [r, g, b].map(x =>
+      x <= 0.03928 ? x / 12.92 : Math.pow((x + 0.055) / 1.055, 2.4)
+    );
+
     return 0.2126 * a[0] + 0.7152 * a[1] + 0.0722 * a[2];
   }
 
+  /**
+   * Parse date from HTML time element.
+   *
+   * @param {string} dateString - The HTML time element string.
+   * @return {Date|null} The parsed date or null if invalid.
+   */
```

Comment:
Good implementation of date parsing from HTML time elements. Consider adding timezone handling if alerts need to respect user timezones:
```javascript
// Consider timezone offset if needed
const date = new Date(match[1]);
// Ensure UTC comparison if dates are stored in UTC
return date;
```